lib/barbeque/message/notification.rb

```diff
@@ -21,7 +21,11 @@ def set_params_from_subscription(subscription)
       def assign_body(message_body)
         super
         @topic_arn = message_body['TopicArn']
-        @body = message_body['Message']
+        @body = if message_body['Message'].is_a?(String)
+                  JSON.parse(message_body['Message'])
+                else
+                  message_body['Message']
```

Comment:
Is this line necessary? If "enqueued via SNS subscriptions, the message comes as JSON-formatted string" is true, the line will be never used. Though I don't understand how it works now.